[DM-3594] OPC-UA reverse scanner NPE fix - changelog (#2387)
SoftwareAG/c8y-docs: content/change-logs/device-management/opcua-10-20-82-0-fix-reverse-scan-edge-case.md

---
date:
title: Fixed OPC-UA reverse address space scanner NullPointerException when last processed node was not root 
product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-3594
version: 10.20.82.0
---

> In previous OPC-UA Gateway versions it was possible to get a NullPointerException during reverse address space scan in rare cases when the last processed node was not a root node.

Reviewer: This desc. with NullPointerException and etc. is very technical. It should be from user perspective. What was not working? Also with this tasks 2 issues are resolved one was relevant to full scan.

Author: done

---
date:

> title: Fixed OPC-UA address space scanning failures in known edge cases 

Reviewer 2: ```suggestion
title: Fixed issue with OPC UA address space scanning
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-3594
version: 10.20.82.0
---

> In earlier versions of the OPC-UA Gateway, there was a rare issue where a reverse address space scan could unexpectedly 

Reviewer 2: ```suggestion
In earlier versions of the OPC UA gateway, it rarely happened that a reverse address space scan unexpectedly 
```

> fail if the last node processed wasn't a root node. We've now fixed this problem. Also, if a scan encounters an issue 

Reviewer 2: ```suggestion
failed if the last node processed wasn't a root node. This issue has been fixed. Also, if a scan encounters an issue 
```
